src/analysis/normed_space/exponential.lean

```diff
@@ -0,0 +1,551 @@
+funext exp_series_sum_eq
+
+lemma exp_eq_tsum_field : exp 𝕂 𝕂 = (λ x : 𝕂, ∑' (n : ℕ), x^n / n!) :=
+funext exp_series_sum_eq_field
+
+lemma exp_zero : exp 𝕂 𝔸 0 = 1 :=
+begin
+  suffices : (λ x : 𝔸, ∑' (n : ℕ), (1 / n! : 𝕂) • x^n) 0 = ∑' (n : ℕ), if n = 0 then 1 else 0,
+  { have key : ∀ n ∉ ({0} : finset ℕ), (if n = 0 then (1 : 𝔸) else 0) = 0,
+    { rintros n hn,
+      rw finset.not_mem_singleton at hn,
+      rw if_neg hn },
+    rw [exp_eq_tsum, this, tsum_eq_sum key, finset.sum_singleton],
+    simp },
+  dsimp only,
+  congr,
+  ext n,
+  split_ifs with h h;
+  simp [h]
+end
+
+section complete_algebra
+
+variables [complete_space 𝔸]
+
+lemma has_fpower_series_on_ball_exp_of_radius_pos (h : 0 < (exp_series 𝕂 𝔸).radius) :
+  has_fpower_series_on_ball (exp 𝕂 𝔸) (exp_series 𝕂 𝔸) 0 (exp_series 𝕂 𝔸).radius :=
+(exp_series 𝕂 𝔸).has_fpower_series_on_ball h
+
+lemma has_fpower_series_at_exp_zero_of_radius_pos (h : 0 < (exp_series 𝕂 𝔸).radius) :
+  has_fpower_series_at (exp 𝕂 𝔸) (exp_series 𝕂 𝔸) 0 :=
+(has_fpower_series_on_ball_exp_of_radius_pos h).has_fpower_series_at
+
+lemma continuous_on_exp :
+  continuous_on (exp 𝕂 𝔸) (emetric.ball 0 (exp_series 𝕂 𝔸).radius) :=
+formal_multilinear_series.continuous_on
+
+lemma analytic_at_exp_of_mem_ball (x : 𝔸) (hx : x ∈ emetric.ball (0 : 𝔸) (exp_series 𝕂 𝔸).radius) :
+  analytic_at 𝕂 (exp 𝕂 𝔸) x:=
+begin
+  by_cases h : (exp_series 𝕂 𝔸).radius = 0,
+  { rw h at hx, exact (ennreal.not_lt_zero hx).elim },
+  { have h := pos_iff_ne_zero.mpr h,
+    exact (has_fpower_series_on_ball_exp_of_radius_pos h).analytic_at_of_mem hx }
+end
+
+/-- The exponential in a Banach-algebra `𝔸` over a normed field `𝕂` has strict Fréchet-derivative
+`1 : 𝔸 →L[𝕂] 𝔸` at zero, as long as it converges on a neighborhood of zero. -/
+lemma has_strict_fderiv_at_exp_zero_of_radius_pos (h : 0 < (exp_series 𝕂 𝔸).radius) :
+  has_strict_fderiv_at (exp 𝕂 𝔸) (1 : 𝔸 →L[𝕂] 𝔸) 0 :=
+begin
+  convert (has_fpower_series_at_exp_zero_of_radius_pos h).has_strict_fderiv_at,
+  ext x,
+  change x = exp_series 𝕂 𝔸 1 (λ _, x),
+  simp [exp_series_apply_eq]
+end
+
+/-- The exponential in a Banach-algebra `𝔸` over a normed field `𝕂` has Fréchet-derivative
+`1 : 𝔸 →L[𝕂] 𝔸` at zero, as long as it converges on a neighborhood of zero. -/
+lemma has_fderiv_at_exp_zero_of_radius_pos (h : 0 < (exp_series 𝕂 𝔸).radius) :
+  has_fderiv_at (exp 𝕂 𝔸) (1 : 𝔸 →L[𝕂] 𝔸) 0 :=
+(has_strict_fderiv_at_exp_zero_of_radius_pos h).has_fderiv_at
+
+/-- In a Banach-algebra `𝔸` over a normed field `𝕂` of characteristic zero, if `x` and `y` are
+in the disk of convergence and commute, then `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`. -/
+lemma exp_add_of_commute_of_lt_radius [char_zero 𝕂]
+  {x y : 𝔸} (hxy : commute x y)
+  (hx : ↑∥x∥₊ < (exp_series 𝕂 𝔸).radius) (hy : ↑∥y∥₊ < (exp_series 𝕂 𝔸).radius) :
+  exp 𝕂 𝔸 (x + y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y) :=
+begin
+  have : ∀ {t : 𝔸}, ↑∥t∥₊ < (exp_series 𝕂 𝔸).radius → summable (λ n : ℕ, ∥(1 / n! : 𝕂) • t ^ n∥),
```

Comment:
No it isn't used anywhere else at the moment. For some reasons I only stated summability lemmas for the R or C case, presumably because I was tired of saying "yes, the radius is infinite", but I'll add general versions.

Reply:
Done (finally)